homeassistant/components/mpd/media_player.py

```diff
@@ -97,75 +99,94 @@ def __init__(self, server, port, password, name):
         self._currentsong = None
         self._playlists = None
         self._currentplaylist = None
-        self._is_connected = False
+        self._is_available = None
         self._muted = False
         self._muted_volume = None
         self._media_position_updated_at = None
         self._media_position = None
         self._media_image_hash = None
         # Track if the song changed so image doesn't have to be loaded every update.
         self._media_image_file = None
-        self._commands = None
 
         # set up MPD client
         self._client = MPDClient()
         self._client.timeout = 30
-        self._client.idletimeout = None
+        self._client.idletimeout = 10
+        self._client_lock = asyncio.Lock()
 
-    async def _connect(self):
-        """Connect to MPD."""
-        try:
-            await self._client.connect(self.server, self.port)
-
-            if self.password is not None:
-                await self._client.password(self.password)
-        except mpd.ConnectionError:
-            return
-
-        self._is_connected = True
-
-    def _disconnect(self):
-        """Disconnect from MPD."""
-        with suppress(mpd.ConnectionError):
-            self._client.disconnect()
-        self._is_connected = False
-        self._status = None
+    @asynccontextmanager
+    async def connection(self):
+        """Handle MPD connect and disconnect."""
```

Comment:
Maybe add a reference explaining the non-persistent connection is a workaround for https://github.com/Mic92/python-mpd2/issues/31